frontend/src/metabase/selectors/app.ts

```diff
@@ -49,6 +49,76 @@ export const getIsAdminApp = createSelector([getRouterPath], path => {
   return path.startsWith("/admin/");
 });
 
+export const getIsCollectionPathVisible = createSelector(
+  [getQuestion, getDashboard, getRouterPath, getIsEmbedded, getEmbedOptions],
+  (question, dashboard, path, isEmbedded, embedOptions) => {
+    if (isEmbedded && !embedOptions.breadcrumbs) {
+      return false;
+    }
+
+    return (
+      ((question != null && question.isSaved()) || dashboard != null) &&
+      PATHS_WITH_COLLECTION_BREADCRUMBS.some(pattern => pattern.test(path))
+    );
+  },
+);
+
+export const getIsQuestionLineageVisible = createSelector(
+  [getQuestion, getOriginalQuestion, getRouterPath],
+  (question, originalQuestion, path) =>
+    question != null &&
+    !question.isSaved() &&
+    originalQuestion != null &&
+    !originalQuestion.isDataset() &&
+    PATHS_WITH_QUESTION_LINEAGE.some(pattern => pattern.test(path)),
+);
+
+export const getIsNavBarEnabled = createSelector(
+  [
+    getUser,
+    getRouterPath,
+    getIsEditingDashboard,
+    getIsEmbedded,
+    getEmbedOptions,
+  ],
+  (currentUser, path, isEditingDashboard, isEmbedded, embedOptions) => {
+    if (!currentUser || isEditingDashboard) {
+      return false;
+    }
+    if (isEmbedded && !embedOptions.side_nav) {
+      return false;
+    }
+    if (isEmbedded && embedOptions.side_nav === "default") {
+      return EMBEDDED_PATHS_WITH_NAVBAR.some(pattern => pattern.test(path));
+    }
+    return !PATHS_WITHOUT_NAVBAR.some(pattern => pattern.test(path));
+  },
+);
+
+const getIsEmbeddedAppBarVisible = createSelector(
```

Comment:
moved some logic from `getIsAppBarVisible`, before adding new logic regarding embedding's `logo` option to make it easier to reason about.